packages/main/src/plugin/extension-loader.ts

```diff
@@ -1074,6 +1078,145 @@ export class ExtensionLoader {
       },
     };
 
+    const navigateTo = (navigateRequest: NavigationRequest) => {
+      this.apiSender.send('navigate', navigateRequest);
+    };
+    const assertContainerIdExist = async (id: string) => {
+      const containers = await containerEngine.listContainers();
+      if (containers.find(container => container.Id === id) === undefined)
+        throw new Error(`Container with id ${id} cannot be found.`);
+    };
+    const assertImageExist = async (id: string, engineId: string, tag: string) => {
+      const images = await containerEngine.listImages();
+      const imageInfo = images.find(c => c.Id === id && c.engineId === engineId);
+      if (imageInfo === undefined) {
+        throw new Error(`Image with id ${id} and engine id ${engineId} cannot be found.`);
+      }
+      if (imageInfo.RepoTags?.find(repoTag => repoTag === tag) === undefined) {
+        throw new Error(`Image with id ${id}, engine id ${engineId} and tag ${tag} cannot be found.`);
+      }
+    };
+    const assertVolumeExist = async (name: string, engineId: string) => {
+      const volumes = await containerProviderRegistry.listVolumes();
+      const allVolumes = volumes.map(volume => volume.Volumes).flat();
+      const matchingVolume = allVolumes.find(volume => volume.Name === name && volume.engineId === engineId);
+      if (matchingVolume === undefined) {
```

Comment:
```suggestion
      if (!matchingVolume) {
```

or use some()

```ts
if (!allVolumes.some(....)
```

Reply:
I think all assert should be part of where it's used

like assertVolumeExist should be moved to containerProviderRegistry class, etc.

extensionLoader should be as small as possible